src/main/scala/com/microsoft/ml/spark/cognitive/TextAnalyticsSDK.scala

```diff
@@ -234,62 +236,29 @@ class TextSentimentV4(override val textAnalyticsOptions: Option[TextAnalyticsReq
           WarningsV4(warnings.getMessage, warnings.getWarningCode.toString)))
     }
 
-    val textSentimentResult = textSentimentResultCollection.asScala.head
-    val documentSentiment = textSentimentResult.getDocumentSentiment();
-
-    val sentimentResult = if (textSentimentResult.isError) {
+    val sentimentResult = textSentimentResultCollection.map(sentiment => if(sentiment.isError) {
       None
     } else {
-      Some(getDocumentSentiment(documentSentiment))
-    }
+      Some(getDocumentSentiment(sentiment.getDocumentSentiment))
+    }).filter(_ != None).toList

```

Comment:
can filter like this way:  collection.filter(sentiment => !sentiment.isError).(--> transfer the filtered results)